app/src/main/java/com/example/tasky/feature_login/presentation/LoginViewModel.kt

```diff
@@ -0,0 +1,42 @@
+package com.example.tasky.feature_login.presentation
+import androidx.lifecycle.ViewModel
+import com.example.tasky.common.domain.isValidEmail
+import com.example.tasky.common.domain.isValidName
+import com.example.tasky.common.domain.isValidPassword
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+
+class LoginViewModel : ViewModel() {
+    private val _name = MutableStateFlow("")
+    val name: StateFlow<String> get() = _name
+
+    private val _email = MutableStateFlow("")
+    val email: StateFlow<String> get() = _email
+
+    private val _password = MutableStateFlow("")
+    val password: StateFlow<String> get() = _password
+
+    private val _isNameValid = MutableStateFlow(false)
+    val isNameValid: StateFlow<Boolean> get() = _isNameValid
+
+    private val _isEmailValid = MutableStateFlow(false)
+    val isEmailValid: StateFlow<Boolean> get() = _isEmailValid
+
+    private val _isPasswordValid = MutableStateFlow(false)
+    val isPasswordValid: StateFlow<Boolean> get() = _isPasswordValid
```

Comment:
Your approach of creating a separate StateFlow for everything has a major downside: It's very error-prone since you always need to remember to also update the isValid states when the underlying text field changes. This becomes really dangerous when state changes from many different places. To fix this, you can work with reactivity which can always be applied when you have two states which always change together (whenever email changes, isEmailValid also needs to be recalculated):
```
val isEmailValid = _email.map { email ->
        email.isValidEmail() // <- Each email emission is mapped to this boolean when it changes
    }.stateIn(viewModelScope, SharingStarted.WhileSubscribed(5000L), false)
```